### PR TITLE
Bs3 rebased

### PR DESCRIPTION
**Pull request by abenari**
replaces #1110
Ready for testing.

### CODE REVIEW

**Comment by dLobatog** [comment]
If you are sure of your changes you can 'git push --force' to avoid closing and opening a new pull request after a rebase, I'll test it too :)

**Comment by dLobatog** [comment]
It looks very sleek! Awesome :+1: 

I especially like how the edit views look now, I haven't noticed any glitches, except maybe for this (see parameters boxes)

http://i.imgur.com/ItQaQ07.png

**Comment by ohadlevy** [comment]
@abenari note, there are test failures

**Comment by abenari** [comment]
[test]

**Comment by ohadlevy** [comment]
btw: interesting to compare http://blog.railsapps.org/post/70815568494/rails-and-bootstrap

File: app/views/layouts/base.html.erb [hunk 1]
```diff
@@ -37,12 +37,8 @@
   </head>
 
   <body>
-    <!--[if lt IE 8]>
-      %p
-        You are using an unsupported browser.
-    <![endif]-->
```
Comment (ehelms) [inline app/views/layouts/base.html.erb:43]: Why get rid of this?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,17 +6,17 @@
     <%= render 'common/notice' unless @notices.empty? %>
     <div id="content" class="container">
       <% unless @page_header.blank? %>
-        <div class="row control-group">
-          <h1 class="span8"><%=h @page_header %></h1>
+        <div class="row  form-group">
```
Comment (ehelms) [inline app/views/layouts/application.html.erb]: Looks like an extra space here.

File: app/views/home/_topbar.html.erb [hunk 3]
```diff
@@ -1,27 +1,40 @@
...
+    <span class="glyphicon glyphicon-cog"></span>
+  </button>
+  <div class="container">
+    <!-- menu -->
+    <%= content_tag :div,nil, :id=>'current_tab', :data=>{:controller=>controller_name, :action=>action_name} %>
+    <% if User.current %>
+      <% cache(TopbarSweeper.fragment_name) do %>
+        <ul class="nav navbar-nav navbar-menu navbar-collapse collapse" id="menu">
+          <%= render "home/org_switcher" %>
+          <%=  render_menu :top_menu %>
```
Comment (ehelms) [inline app/views/home/_topbar.html.erb]: Looks like an extra space here.

File: app/views/home/_topbar.html.erb [hunk 4]
```diff
@@ -1,27 +1,40 @@
...
+        </ul>
+        <ul class="nav navbar-nav navbar-admin navbar-right navbar-collapse collapse" id="menu2">
+          <%=  render_menu :admin_menu %>
```
Comment (ehelms) [inline app/views/home/_topbar.html.erb]: Looks like an extra space here.

**Comment by ohadlevy** [comment]
@abenari would it make more sense to use the 

``` ruby
gem 'bootstrap-sass', '~> 3.0.3.0'
```

it seems like this is the new official version for bootstrap rails gem

**Comment by ohadlevy** [comment]
when updating parameters, the size, location seems wrong, also when trying to resize a parameter value text area, it all moves around

**Comment by waldenraines** [comment]
@ohadlevy that should be fine I think since the Foreman changes are primarily css class names, markup changes.  I used `anjlab-bootstrap-rails` because I couldn't find any bootstrap 3 gems out there when I originally did this work.  The plan was always to replace it with an official gem.

**Comment by skottler** [comment]
One kind of annoying aesthetic thing is that the text doesn't sit properly inside tables, there's more padding on the bottom of the cell than on the top:

tabular

**Comment by waldenraines** [comment]
@skottler nice catch!

Adding `vertical-align: middle;` to `.table td` seems to fix that.

**Comment by ehelms** [comment]
screenshot from 2013-12-23 15 49 59

I noticed the above for both of the flyouts. There is a break in the top border of the flyout as well as the shadowing and placement appear off.

**Comment by ohadlevy** [comment]
is it just me, or the forms are messed up a bit?:

``` diff
diff --git a/app/helpers/layout_helper.rb b/app/helpers/layout_helper.rb
index a93a3e4..b616385 100644
--- a/app/helpers/layout_helper.rb
+++ b/app/helpers/layout_helper.rb
@@ -136,13 +136,11 @@ module LayoutHelper
       label   = options.delete(:label)
       label ||= ((clazz = f.object.class).respond_to?(:gettext_translation_for_
                   s_(clazz.gettext_translation_for_attribute_name attr)) if f
-      label_tag(attr, label, :class=>"control-label").html_safe +
-        content_tag(:div, :class => "controls") do
-          content_tag(:div, :class => size_class) do
-            yield.html_safe + help_block.html_safe
-          end.html_safe + help_inline.html_safe
-        end.html_safe
-    end
+      label_tag(attr, label, :class=>"col-sm-2 control-label").html_safe +
+        content_tag(:div, :class => size_class) do
+          yield.html_safe + help_block.html_safe
+        end.html_safe + help_inline.html_safe
+      end.html_safe
     end
```

**Comment by ohadlevy** [comment]
also, fix for 1.87

``` diff
diff --git a/app/helpers/layout_helper.rb b/app/helpers/layout_helper.rb
index a93a3e4..a9dbefc 100644
--- a/app/helpers/layout_helper.rb
+++ b/app/helpers/layout_helper.rb
@@ -24,7 +24,7 @@ module LayoutHelper
     content_for(:javascripts) { javascript_include_tag(*args) }
   end

-  def addClass options={}, new_class
+  def addClass(options = {}, new_class = nil)
```

**Comment by ohadlevy** [comment]
pagination html is broken when using the js Table (such as vm listing under compute resources)
screenshot from 2013-12-25 15 39 28

**Comment by ohadlevy** [comment]
also, fact graphs seems to be broken
screenshot from 2013-12-25 15 51 43

**Comment by ohadlevy** [comment]
popup locations are at times wrong, see two examples
screenshot from 2013-12-25 15 55 09

**Comment by ohadlevy** [comment]
inline editing for setting text seems wrong too:
screenshot from 2013-12-25 15 57 57

**Comment by ohadlevy** [comment]
also, the about status badges seems to lost the color? not sure if its on purpose

**Comment by abenari** [comment]
[test]

**Comment by ohadlevy** [comment]
merged as bf4a13d d31eea8 and c9aca07
Thanks @waldenraines @abenari and @domcleal

**Pull request closed without merging**